tests/test_load.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+
+from substratest import assets
+
+# COMPUTE_PLAN_SIZES = [10, 100, 1000, 10000]
+COMPUTE_PLAN_SIZES = [100]
+
+
+@pytest.mark.parametrize('compute_plan_size', COMPUTE_PLAN_SIZES)
+def test_load_single_node_1_branch(compute_plan_size, global_execution_env):
+    """
+    Shape of the compute plan:
+
+    Node A: traintuple 0 -> traintuple -> ... -> traintuple N
+    """
+    factory, network = global_execution_env
+    session = network.sessions[0].copy()
+
+    dataset = session.state.datasets[0]
+
+    spec = factory.create_algo()
+    algo = session.add_algo(spec)
+
+    cp_spec = factory.create_compute_plan()
+    previous_tuple = None
+    for _ in range(compute_plan_size):
+        previous_tuple = cp_spec.add_traintuple(
+            algo=algo,
+            dataset=dataset,
+            data_samples=[dataset.train_data_sample_keys[0]],
+            in_models=[previous_tuple] if previous_tuple else None,
+        )
+
+    cp = session.add_compute_plan(cp_spec)
+    first_tuple = cp.list_traintuple()[0]
+    assert first_tuple.rank == 0
+    first_tuple = first_tuple.future().wait()
+    assert first_tuple.status == assets.Status.done
+
+    cp = session.cancel_compute_plan(cp.compute_plan_id)
+    assert cp.status == assets.Status.canceled
+
+    cp = cp.future().wait()
+    assert cp.status == assets.Status.canceled
+
+    first_tuple = cp.list_traintuple()[0]
+    assert first_tuple.rank == 0
+    assert first_tuple.status == assets.Status.done
+
+
+@pytest.mark.parametrize('compute_plan_size', COMPUTE_PLAN_SIZES)
+def test_load_multi_node_2_branches(compute_plan_size, global_execution_env):
+    """
+    Shape of the compute plan:
+
+    Node A: traintuple 0 -> traintuple -> ... -> traintuple
+    Node B:              -> traintuple -> ... -> traintuple
+    """
+
+    factory, network = global_execution_env
+    session_1 = network.sessions[0].copy()
+    session_2 = network.sessions[0].copy()
```

Comment:
```suggestion
    session_2 = network.sessions[1].copy()
```

Reply:
Thank you! That was a big typo. I fixed it everywhere.